### DISCUSSION TITLE
[nodejs] Behavior of Vercel after using res.send();

### DISCUSSION THREAD

**Discussion by vvo** [vercel/vercel#4949] [Help] [answered]
Hi there, let's say I do this:

```js
export async function handler(req, res) {
  res.send({ok: true});
  someAsyncFunction();
}
```

Is someAsyncFunction(); guaranteed to complete or will Vercel/Aws/GCP kill it before it finishes?

I'd like to get a comprehensive explanation/documentation on the various behaviours, for example this one also is unclear:

```js
export async function handler(req, res) {
  res.send({ok: true});
  await someAsyncFunction();
}
```

And (similar to the first one):

```js
export async function handler(req, res) {
  someAsyncFunction();
  res.send({ok: true});
}
```

It would be nice to know those limits/get recommendations from the Vercel team.

Related readings:
- https://levelup.gitconnected.com/avoiding-the-pitfalls-of-async-node-js-functions-in-aws-lambda-941220582e7a
- https://lumigo.io/blog/node-js-lambda-execution-leaks-a-practical-guide/

**Comment by styfle** [comment, marked as answer]
The "fire-and-forget" pattern is not supported as documented on the Limits page.

https://vercel.com/docs/v2/platform/limits#streaming-responses

You will need to await the async task: `await someAsyncFunction()`

**Reply by vvo** [reply]
Thanks, very clear 🙏

**Reply by faceyspacey** [reply]
It would be crystal clear if it was described that calling `await someAsyncFunction()` after `res.send` does not work either, which is very unusual behavior outside of the serverless world.

Lambda seems to support the ability to call a `callback` when done. I'm not sure why Vercel doesn't provide an option to signal the *real* end of the function.
